core/src/main/clojure/xtdb/expression/metadata.clj

```diff
@@ -27,32 +28,50 @@
   (letfn [(var-value-expr [f meta-value field value-type value-expr]
             ;; TODO adapt for boolean metadata writer
             (when-not (contains? bool-metadata-types value-type)
-              (let [base-col-types (-> (get col-types field)
-                                       types/flatten-union-types)]
-                (simplify-and-or-expr
-                 {:op :call
-                  :f :or
-                  ;; TODO this seems like it could make better use
-                  ;; of the polymorphic expr patterns?
-                  :args (vec
-                         (for [col-type (cond
-                                          (isa? types/col-type-hierarchy value-type :num)
-                                          (filterv types/num-types base-col-types)
-
-                                          (and (vector? value-type) (isa? types/col-type-hierarchy (first value-type) :date-time))
-                                          (filterv (comp types/date-time-types types/col-type-head) base-col-types)
-
-                                          (contains? base-col-types value-type)
-                                          [value-type])]
-
-                           {:op :test-metadata,
-                            :f f
-                            :meta-value meta-value
-                            :col-type col-type
-                            :field field,
-                            :value-expr value-expr
-                            :bloom-hash-sym (when (= meta-value :bloom-filter)
-                                              (gensym 'bloom-hashes))}))}))))
+              (let [base-col-types (->> (get col-types field)
+                                        types/flatten-union-types)]
+
+                (if (= meta-value :bloom-filter)
```

Comment:
I only updated this bit for bloom filters. I don't know if the min and max logic should also be updated to do something smarter with regards to num types hierarchy or date-time hierarchy. Maybe one for another ticket.

Reply:
yes, if we're converting (e.g.) numbers to doubles for hashing we should probably use the same process for inequalities. let's figure out how much work it is to make the decision of whether to include it here or not?

Reply:
Adding an extra card for min/max/bloom logic cleanup in metadata/expression.clj.

Reply:
Carded up at https://github.com/xtdb/xtdb/issues/4116.